### PR TITLE
[Enhancement](Doe) Be query es use fe generate dsl. 

### LINKED ISSUE #11804: [Enhancement](Doe) Be query es use fe generate dsl.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Be query es use fe generate dsl.

### Solution

Remove be generate dsl logic, use fe dsl.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close https://github.com/apache/doris/issues/11804

## Problem summary

1. Be query es use fe generate dsl. 
    - use fe config `enable_new_es_dsl` control whether enabled.
    - for some functions that can't be pushed down, doris filter
3. fix cast error

```
+---------+-------------+--------------------+------------+
| test1   | test2       | test3              | test4      |
+---------+-------------+--------------------+------------+
| test1   | hello world | 3.1414999961853027 | 2022-08-08 |
| test2   | hello world | 2.4149999618530273 | 2022-08-08 |
| stalary | test1       | 3.1410000324249268 | 2022-08-08 |
+---------+-------------+--------------------+------------+
```

old_query
```
mysql> select * from test_es7 where cast(test3 as int)=3;
Empty set (0.01 sec)
```

new_query
```
mysql> select * from test_es7 where cast(test3 as int)=3;
+---------+-------------+--------------------+------------+
| test1   | test2       | test3              | test4      |
+---------+-------------+--------------------+------------+
| test1   | hello world | 3.1414999961853027 | 2022-08-08 |
| stalary | test1       | 3.1410000324249268 | 2022-08-08 |
+---------+-------------+--------------------+------------+
2 rows in set (0.51 sec)

mysql> select * from test_es7 where cast(test3 as int)=2;
+-------+-------------+--------------------+------------+
| test1 | test2       | test3              | test4      |
+-------+-------------+--------------------+------------+
| test2 | hello world | 2.4149999618530273 | 2022-08-08 |
+-------+-------------+--------------------+------------+
1 row in set (0.03 sec)

mysql> select * from test_es7 where cast(test3 as int)=1;
Empty set (0.05 sec)
```

old_explain
```
mysql> explain select * from test_es7 where cast(test3 as int)=3;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Explain String                                                                                                                                                                               |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                                                                                                                                              |
|   OUTPUT EXPRS:`default_cluster:test`.`test_es7`.`test1` | `default_cluster:test`.`test_es7`.`test2` | `default_cluster:test`.`test_es7`.`test3` | `default_cluster:test`.`test_es7`.`test4` |
|   PARTITION: RANDOM                                                                                                                                                                          |
|                                                                                                                                                                                              |
|   RESULT SINK                                                                                                                                                                                |
|                                                                                                                                                                                              |
|   0:EsScanNode                                                                                                                                                                               |
|      TABLE: test_es7                                                                                                                                                                         |
|      PREDICATES: CAST(`test3` AS INT) = 3                                                                                                                                                    |
|      LOCAL_PREDICATES:                                                                                                                                                                       |
|      REMOTE_PREDICATES:                                                                                                                                                                      |
|      ES_QUERY_DSL:                                                                                                                                                                           |
|      ES index/type: test/null                                                                                                                                                                |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

new_explain
```
mysql> explain select * from test_es7 where cast(test3 as int)=3;
+-------------------------------------------------+
| Explain String                                  |
+-------------------------------------------------+
| PLAN FRAGMENT 0                                 |
|   OUTPUT EXPRS:                                 |
|     `default_cluster:test`.`test_es7`.`test1`   |
|     `default_cluster:test`.`test_es7`.`test2`   |
|     `default_cluster:test`.`test_es7`.`test3`   |
|     `default_cluster:test`.`test_es7`.`test4`   |
|   PARTITION: RANDOM                             |
|                                                 |
|   VRESULT SINK                                  |
|                                                 |
|   0:VEsScanNode                                 |
|      TABLE: test_es7                            |
|      LOCAL_PREDICATES: CAST(`test3` AS INT) = 3 |
|      REMOTE_PREDICATES: {"match_all":{}}        |
|      ES index/type: test/null                   |
+-------------------------------------------------+
```

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
4. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
5. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java [hunk 1]
```diff
@@ -369,6 +372,9 @@ private void buildQuery() {
             } else {
                 queryBuilder = boolQueryBuilder;
             }
+            if (Config.enable_new_es_dsl) {
+                notPushDownList.forEach(expr -> conjuncts.removeIf(e -> e.equals(expr)));
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java:376]: should we remove expression which is **not** in `notPushDownList` instead?

File: be/src/vec/exec/ves_http_scan_node.cpp [hunk 2]
```diff
@@ -124,6 +124,12 @@ Status VEsHttpScanNode::open(RuntimeState* state) {
     SCOPED_CONSUME_MEM_TRACKER(mem_tracker());
     RETURN_IF_CANCELLED(state);
 
+    std::map<std::string, std::string> properties(_properties);
+    if (properties.find(ESScanReader::KEY_QUERY_DSL) != properties.end()) {
```
Comment (Gabriel39) [inline be/src/vec/exec/ves_http_scan_node.cpp:128]: Why not using `_properties` directly?

Comment (Gabriel39) [reply be/src/vec/exec/ves_http_scan_node.cpp:128]: a little confused. Maybe you should add a comment here to point out `KEY_QUERY_DSL` in properties means we turn `enable_new_es_dsl` on in FE and we have already removed push-down predicates here.

Comment (stalary) [reply be/src/vec/exec/ves_http_scan_node.cpp:128]: > a little confused. Maybe you should add a comment here to point out `KEY_QUERY_DSL` in properties means we turn `enable_new_es_dsl` on in FE and we have already removed push-down predicates here.

In order to pass only one field, the argument logic is controlled in FE, so let me comment it.

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**